opencl-aot/source/main.cpp

```diff
@@ -414,7 +414,14 @@ int main(int Argc, char *Argv[]) {
   CLProgramUPtr ProgramUPtr(std::move(Progs[0]));
 
   // step 7: set OpenCL build options
-  std::string BuildOptions = OptBuildOptions;
+  std::string BuildOptsStr;
```

Comment:
Better to use `BuildOptions` variable here, the number of variables reduced.

Reply:
I thought so too, but on [433 line](https://github.com/intel/llvm/pull/2828/files#diff-ce4bb7de20bfd3d99501195c52a715d758b2e79d70372baa5952ae32a92b4fb2R433) we use the initial build options string, not the `BuildOptions` with the parent directory path attached (on line 427). And we, unfortunately, can't reuse `OptBuildOptions` for this purpose as it was done previously

Reply:
Oh, it looks like a bug on my side - ```BuildOptions``` should have been passed to ```clBuildProgram```, not ```OptBuildOptions```. That's because on line 427 a mandatory option ```-I"/path/to/dir"``` is adding to ```BuildOptions```, and after that it prints to cout, and it doesn't use in any way after that - just another variable - ```OptBuildOptions``` - is used in algorithm below. This mandatory option is required for FPGA Emu, and dpcpp don't use opencl-aot for AOT FPGA Emu, but definitely is should be fixed. 
Can you please try to use the single variable ```BuildOptions``` here?

Reply:
Oh, that makes more sense. 
Sure, I've uploaded the change you asked for, you're welcome to review it.